logs/logs.go

```diff
@@ -52,17 +52,19 @@ type LogDest interface {
 
 // LogAgent is the agent handles pure log pipelines
 type LogAgent struct {
-	Config      *config.Config
-	backends    map[string]LogBackend
-	destNames   map[LogDest]string
-	collections []LogCollection
+	Config              *config.Config
+	backends            map[string]LogBackend
+	destNames           map[LogDest]string
+	collections         []LogCollection
+	retentionAlreadySet map[string]bool
 }
 
 func NewLogAgent(c *config.Config) *LogAgent {
 	return &LogAgent{
-		Config:    c,
-		backends:  make(map[string]LogBackend),
-		destNames: make(map[LogDest]string),
+		Config:              c,
+		backends:            make(map[string]LogBackend),
+		destNames:           make(map[LogDest]string),
+		retentionAlreadySet: make(map[string]bool),
```

Comment:
**Nit** Should we better to use struct{} instead of bool here @SaxyPandaBear ?

Reply:
It doesn't make enough difference for me to suggest changing it. Changing this to a `map[string]struct{}` sacrifice what amounts to a couple bytes of memory used to hold a boolean in a map for readability. 
```go
m := make(map[string]struct{})
m["a"] = struct{}{}
if _, ok := m["b"]; ok {
    fmt.Println("hello")
} else if _, ok := m["a"]; ok {
    fmt.Println("hello there")
} else {
    fmt.Println("bye")
}
```
Switching to a `struct{}` complicates the code.